### PR TITLE
Directory api

### PR DESCRIPTION
**Pull request by tych0**

### CODE REVIEW

**Comment by stgraber** [comment]
Code looks good to me.

Can you also update doc/rest-api.md to mention that /files also works for directories and to mention the new header?

And it'd probably be nice to have a test that checks the file modes and ownership after pull and after push since we've ran into a number of problems around that before.

**Comment by stgraber** [comment]
Also, interesting Jenkins failure which looks like a golang bug (blew up in pthread :)).

File: lxc/file.go [hunk 1]
```diff
@@ -32,17 +36,30 @@ func (c *fileCmd) usage() string {
...
+To pull /etc/hosts from the container:
+  lxc file pull foo/etc/hosts .
+`)
 }
 
 func (c *fileCmd) flags() {
 	gnuflag.IntVar(&c.uid, "uid", -1, i18n.G("Set the file's uid on push"))
 	gnuflag.IntVar(&c.gid, "gid", -1, i18n.G("Set the file's gid on push"))
 	gnuflag.StringVar(&c.mode, "mode", "", i18n.G("Set the file's perms on push"))
+	gnuflag.BoolVar(&c.recursive, "recusrive", false, i18n.G("Recursively push or pull files"))
```
Comment (sean-jc) [inline lxc/file.go]: The name of the flag has a typo.

Comment (tych0) [inline lxc/file.go]: Ah ha, so it does, thanks :)

**Pull request merged**